ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -1220,17 +1220,22 @@ private class JdbcLedgerDao(
     ~ str("observers").? map flatten)
 
   private val ContractLetParser = (ledgerString("id")
-    ~ date("effective_at").? map flatten)
+    ~ date("effective_at").?
+    ~ long("archive_offset").? map flatten)
 
   private val SQL_SELECT_CONTRACT =
     SQL(queries.SQL_SELECT_CONTRACT)
 
   private val SQL_SELECT_CONTRACT_LET =
     SQL("""
-        |select c.id, le.effective_at
-        |from contracts c
+        |select cd.id, le.effective_at
+        |from contract_data cd
+        |left join contracts c on c.id=cd.id
         |left join ledger_entries le on c.transaction_id = le.transaction_id
-        |where c.id={contract_id} and c.archive_offset is null""".stripMargin)
+        |where
+        |  cd.id={contract_id} and
+        |  ((le.effective_at is not null and c.archive_offset is null) or le.effective_at is null)
```

Comment:
This could be `cd.id={contract_id} and (le.effective_at is null || c.archive_offset is null)`.

Reply:
:+1: